design/api.go

```diff
@@ -115,6 +115,9 @@ func (a *APIExpr) Random() *Random {
 // EvalName is the qualified name of the expression.
 func (a *APIExpr) EvalName() string { return "API " + a.Name }
 
+// Hash returns a unique hash value for a.
+func (a *APIExpr) Hash() string { return a.Name }
```

Comment:
We may need to specify a prefix here so there's no risk of clashing with other constructs.